source/lib/cdk-textract-stack.ts

```diff
@@ -255,21 +256,156 @@ export class CdkTextractStack extends cdk.Stack {
+      {
+        accessControl: s3.BucketAccessControl.LOG_DELIVERY_WRITE,
+        versioned: false,
+        encryption: BucketEncryption.S3_MANAGED,
+        blockPublicAccess: BlockPublicAccess.BLOCK_ALL,
+      }
+    );
+
+    const edgeCodeDeployment = new s3deploy.BucketDeployment(
+      this,
+      this.resourceName("EdgeLambdaDeployment"),
+      {
+        sources: [s3deploy.Source.asset("lambda/edge")],
+        destinationBucket: edgeLambdaCodeBucket,
+      }
+    );
+
+    const egdeLambdaRole = new iam.Role(
+      this,
+      this.resourceName("EdgeLambdaServiceRole"),
+      {
+        assumedBy: new iam.CompositePrincipal(
+          new iam.ServicePrincipal("lambda.amazonaws.com"),
+          new iam.ServicePrincipal("edgelambda.amazonaws.com")
+        ),
+        inlinePolicies: {
+          BasicExecution: new iam.PolicyDocument({
+            assignSids: true,
+            statements: [
+              new iam.PolicyStatement({
+                actions: [
+                  "logs:CreateLogGroup",
+                  "logs:CreateLogStream",
+                  "logs:PutLogEvents",
+                ],
+                resources: ["*"],
+              }),
+            ],
+          }),
+        },
+      }
+    );
+    
+    const onEventEdgeLambdaCreator = new lambda.Function(
+      this,
+      this.resourceName("onEventEdgeLambdaCreator"),
+      {
+        code: lambda.Code.fromAsset("lambda/customResourceEdgeLambdaCreator/"),
+        description: "onEvent handler for creating Edge Lambda",
+        runtime: lambda.Runtime.PYTHON_3_8,
+        handler: "lambda_function.lambda_handler",
+        timeout: Duration.minutes(15),
+        environment: {
+          EDGE_LAMBDA_NAME: this.resourceName("DUSEdgeLambda"),
+          EDGE_LAMBDA_ROLE_ARN: egdeLambdaRole.roleArn,
+          SOURCE_KEY: "lambda_function.py",
+          CLOUDFRONT_DIST_ID: distribution.distributionId,
+          BUCKET_NAME: edgeLambdaCodeBucket.bucketName,
+        },
+      }
+    );
+
+    onEventEdgeLambdaCreator.node.addDependency(edgeCodeDeployment);
+
+    onEventEdgeLambdaCreator.addToRolePolicy(
+      new iam.PolicyStatement({
+        effect: iam.Effect.ALLOW,
+        actions: ["lambda:CreateFunction", "lambda:DeleteFunction", "lambda:GetFunction"],
+        resources: ["arn:aws:lambda:us-east-1:" + this.account + ":*"],
+      })
+    );
+
+    onEventEdgeLambdaCreator.addToRolePolicy(
+      new iam.PolicyStatement({
+        effect: iam.Effect.ALLOW,
+        actions: ["iam:PassRole"],
+        resources: [egdeLambdaRole.roleArn],
+      })
+    );
+
+    onEventEdgeLambdaCreator.addToRolePolicy(
+      new iam.PolicyStatement({
+        effect: iam.Effect.ALLOW,
+        actions: ["s3:Get*", "s3:List*"],
+        resources: [
+          edgeLambdaCodeBucket.bucketArn,
+          `${edgeLambdaCodeBucket.bucketArn}/*`,
+        ],
+      })
+    );
+
+    onEventEdgeLambdaCreator.addToRolePolicy(
```

Comment:
This may be an issue for Solutions Builder. Is it possible to scope out the permission to the specific Cfn Distribution?

Reply:
unfortunately not. Cloudfront is one of those services that do not support more granular resources than * right now